src/lib.rs

```diff
@@ -1,19 +1,322 @@
+
+/// Minimum value of an object of type [`int8_t`].
+pub const INT8_MIN: int8_t = internal::INT8_MIN as _;
+
+/// Maximum value of an object of type [`int8_t`].
+pub const INT8_MAX: int8_t = internal::INT8_MAX as _;
+
+/// Maximum value of an object of type [`uint8_t`].
+pub const UINT8_MAX: uint8_t = internal::UINT8_MAX as _;
+
+/// Minimum value of an object of type [`int16_t`].
+pub const INT16_MIN: int16_t = internal::INT16_MIN as _;
+
+/// Maximum value of an object of type [`int16_t`].
+pub const INT16_MAX: int16_t = internal::INT16_MAX as _;
+
+/// Maximum value of an object of type [`uint16_t`].
+pub const UINT16_MAX: uint16_t = internal::UINT16_MAX as _;
+
+/// Minimum value of an object of type [`int32_t`].
+pub const INT32_MIN: int32_t = internal::INT32_MIN as _;
+
+/// Maximum value of an object of type [`int32_t`].
+pub const INT32_MAX: int32_t = internal::INT32_MAX as _;
+
+/// Maximum value of an object of type [`uint32_t`].
+pub const UINT32_MAX: uint32_t = internal::UINT32_MAX as _;
+
+/// Minimum value of an object of type [`int64_t`].
+pub const INT64_MIN: int64_t = internal::_INT64_MIN as _;
+
+/// Maximum value of an object of type [`int64_t`].
+pub const INT64_MAX: int64_t = internal::_INT64_MAX as _;
+
+/// Maximum value of an object of type [`uint64_t`].
+pub const UINT64_MAX: uint64_t = uint64_t::MAX as _;
+
+/// Minimum value of an object of type [`int_fast8_t`].
+pub const INT_FAST8_MIN: int_fast8_t = internal::INT_FAST8_MIN as _;
+
+/// Maximum value of an object of type [`int_fast8_t`].
+pub const INT_FAST8_MAX: int_fast8_t = internal::INT_FAST8_MAX as _;
+
+/// Maximum value of an object of type [`uint_fast8_t`].
+pub const UINT_FAST8_MAX: uint_fast8_t = internal::UINT_FAST8_MAX as _;
 
-pub use self::internal::*;
+/// Minimum value of an object of type [`int_fast16_t`].
+pub const INT_FAST16_MIN: int_fast16_t = internal::INT_FAST16_MIN as _;
+
+/// Maximum value of an object of type [`int_fast16_t`].
+pub const INT_FAST16_MAX: int_fast16_t = internal::INT_FAST16_MAX as _;
+
+/// Maximum value of an object of type [`uint_fast16_t`].
+pub const UINT_FAST16_MAX: uint_fast16_t = internal::UINT_FAST16_MAX as _;
+
+/// Minimum value of an object of type [`int_fast32_t`].
+pub const INT_FAST32_MIN: int_fast32_t = internal::INT_FAST32_MIN as _;
+
+/// Maximum value of an object of type [`int_fast32_t`].
+pub const INT_FAST32_MAX: int_fast32_t = internal::INT_FAST32_MAX as _;
+
+/// Maximum value of an object of type [`uint_fast32_t`].
+pub const UINT_FAST32_MAX: uint_fast32_t = internal::UINT_FAST32_MAX as _;
+
+/// Minimum value of an object of type [`int_fast64_t`].
+pub const INT_FAST64_MIN: int_fast64_t = internal::_INT_FAST64_MIN as _;
+
+/// Maximum value of an object of type [`int_fast64_t`].
+pub const INT_FAST64_MAX: int_fast64_t = internal::_INT_FAST64_MAX as _;
+
+/// Maximum value of an object of type [`uint_fast64_t`].
+pub const UINT_FAST64_MAX: uint_fast64_t = uint64_t::MAX as _;
```

Comment:
I believe this should be `_UINT_FAST64_MAX`, no? The constant is otherwise unused...
Dtto a few more of those consts...

Reply:
Ah, I need to remove that one. I planned on using it but bindgen sadly makes this an extern "C" static rather than a const.

Reply:
Huh, that's unfortunate. Technically this shouldn't be set to `uint64_t::MAX` because it may in theory by larger. (I doubt there's actually an implementation where that would be true, but still, we should adhere to the standard.)

Reply:
Ah,y es - `uint64_t::MAX` is a typo, it should be `uint_fast64_t::MAX`. Fixed that. I imagine one way to get the actual value over is either a function, but then we do have to statically link, or a string that's statically parsed on the Rust side. That seemed odd to me to, particularly since the value is mapped to a Rust type after all.